src/lib/error_logger.py

```diff
@@ -52,14 +55,59 @@ class ErrorLogger:
     Simple class to be inherited by other classes to add error logging functions.
     """
 
-    logging.basicConfig(format="%(message)s")
+    name: str
+    """ Name of the logger, defaults to the class name. """
+
+    logger: logging.Logger
+    """ Instance of logger which will be used. Each ErrorLogger instance has its own Logger. """
+
+    def __init__(self, name: str = None):
+
+        # Default to the classname
+        self.name = name or self.__class__.__name__
+
+        # Create an instance of logger
+        self.logger = logging.getLogger(self.name)
+
+        # Read logging level from env variable, default to INFO
+        logging_level = {
+            "DEBUG": logging.DEBUG,
+            "INFO": logging.INFO,
+            "WARNING": logging.WARNING,
+            "ERROR": logging.ERROR,
+        }.get(os.getenv("LOG_LEVEL"), logging.INFO)
+        self.logger.setLevel(logging_level)
+
+        # Configure the handler to use our preferred logging format
+        handler = logging.StreamHandler()
+        handler.setFormatter(logging.Formatter("%(message)s"))
+        self.logger.addHandler(handler)
 
     def timestamp(self) -> str:
         return datetime.datetime.now().isoformat()[:24]
 
-    def errlog(self, msg: str, **kwargs) -> None:
-        logging.warning(
+    def _log_msg(self, log_func: Callable, msg: str, **kwargs) -> None:
+        log_func(
             StructuredMessage(
-                msg, timestamp=self.timestamp(), classname=self.__class__.__name__, **kwargs
+                msg,
+                logname=self.name,
+                timestamp=self.timestamp(),
+                # TODO: consider whether we should keep classname or if logname is sufficient
+                classname=self.__class__.__name__,
+                **kwargs,
             )
         )
+
+    def log_error(self, msg: str, **kwargs) -> None:
+        # TODO: replace logging.warning with logging.error once we are sure it will not break
```

Comment:
I'm fairly confident that this won't break the monitoring infrastructure - in the sense that the dashboards and filters are based on all of these going to stderr anyway.

Reply:
The move to json based logging meant some issues aren't automatically caught by the cloud error reporting tool (and thus our infra). However it made monitoring, charting and alerting much easier. One way to fix this and have the best of both worlds would be to use the cloud API to log errors directly with the error reporting tool.

Reply:
The automated error reporting based on stderr prior to JSON logging wasn't super useful, I think what you set up is much better than the default behavior. I want to double down on that and do structured logging for events beyond errors, so we can parse logs to look for abnormal behavior which may not always result in an error.

If there is no distinction between warning and error on the Cloud side, then I'll log the appropriate level here.